solutions/devsprint-leonardo-santos-3/FinanceAppTests/Services/FinanceServiceTests.swift

```diff
@@ -45,6 +42,42 @@ final class FinanceServiceTests: XCTestCase {
         makeDefaultTests()
     }
     
+    func test_fetchHomeData_shouldPassCorrectURL(){
+        sut.fetchHomeData({ _ in })
+        
+        XCTAssertEqual(networkClientSpy.url?.description, "https://raw.githubusercontent.com/devpass-tech/challenge-finance-app/main/api/home_endpoint.json")
+        XCTAssertEqual(networkClientSpy.performRequestCount, 1)
+    }
+    
+    func test_fetchHomeData_givenNilData_shouldReturnNil(){
+        networkClientSpy.completionData = nil
+        
+        sut.fetchHomeData({
+            XCTAssertNil($0)
+            XCTAssertEqual(self.networkClientSpy.performRequestCount, 1)
+        })
+    }
+    
+    func test_fetchHomeData_givenValidData_shouldReturnHomeData(){
+        let expectedResult: HomeData = HomeData.fixture(balance: 100.0)
+        
+        networkClientSpy.completionData = correctHomeDataData
+        
+        sut.fetchHomeData({
+            XCTAssertEqual($0?.balance, expectedResult.balance)
```

Comment:
Aqui podemos passar diretamente o resultado esperado se criarmos a `expectedResult` já com os mesmos valores  do `correctHomeData`, o que acha? 

```suggestion
            XCTAssertEqual($0, expectedResult)
```